extension/extension_config.cmake

```diff
@@ -8,3 +8,15 @@
 #add_static_link_extension(sqlite)
 #add_static_link_extension(unity_catalog)
 #add_static_link_extension(vector)
+
+if(${BUILD_WASM})
```

Comment:
Simply write as `if(BUILD_WASM)`?  if statement in cmake should evaluate BUILD_WASM, so you don't need ${} around it